src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,607 @@
+                    if [i for i in list_items(x.rsplit('/',1)[0]) if i.d_name.decode("utf-8") == x.rsplit('/',1)[1] and  i.is_dir()]:
+                        if giveDirs:
+                            yield x
+                    else:
+                        yield x
+        else:
+            yield fullpath.replace('//', '/')
+
+class CephfsShell(Cmd):
+
+    def __init__(self):
+        super().__init__()
+        self.set_prompt()
+        self.intro = 'Ceph File System Shell'
+        self.interactive = False
+        
+    def default(self, line):
+        print("Unrecognized command:", line)
+
+    def set_prompt(self):
+        self.prompt = '\033[01;33mCephFS >>>\033[00m ' 
+   
+    def create_argparser(self, command):
+        try:
+            argparse_args = getattr(self, "argparse_" + command)
+        except AttributeError:
+            return None
+        doc_lines = getattr(self, "do_" + command).__doc__.expandtabs().splitlines()
+        if '' in doc_lines:
+            blank_idx = doc_lines.index('')
+            usage = doc_lines[:blank_idx]
+            description = doc_lines[blank_idx + 1:]
+        else:
+            usage = doc_lines
+            description = []
+        parser = argparse.ArgumentParser(
+            prog=command,
+            usage='\n'.join(usage),
+            description='\n'.join(description),
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        for args, kwargs in argparse_args:
+            parser.add_argument(*args, **kwargs)
+        return parser
+
+    def complete_filenames(self, text, line, begidx, endidx): 
+        if not text:
+            completions = [x.d_name.decode("utf-8") + '/'*int(x.is_dir()) for x in list_items(cephfs.getcwd())]
+        else:
+            if text.count('/') > 0:
+                completions = [text.rsplit('/',1)[0] +'/'+ x.d_name.decode("utf-8") + '/'*int(x.is_dir())  for x in list_items('/' + text.rsplit('/',1)[0]) if x.d_name.decode("utf-8").startswith(text.rsplit('/',1)[1])]
+            else :
+                completions = [x.d_name.decode("utf-8") + '/'*int(x.is_dir()) for x in list_items() if x.d_name.decode("utf-8").startswith(text)]
+        return completions
+    
+    def onecmd(self, line):
+        """Global error catcher"""
+        try:
+            res = Cmd.onecmd(self, line)
+            if self.interactive: 
+                self.set_prompt()
+            return res
+        except ConnectionError as e:
+            print("***", e)
+        except KeyboardInterrupt:
+            print("Command aborted")
+        except Exception as e:
+            print(e)
+            traceback.print_exc(file=sys.stdout)
+
+    def complete_mkdir(self, text, line, begidx, endidx):
+        """ auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    mkdir_parser =  argparse.ArgumentParser(description='Create Directory.')
```

Comment:
In my opinion all arguments should be defined somewhere at the top of the file and not distributed accross the whole file.